acidlake/nolatech-backend-api
# 3-backend-project-initial-setup

## Description:
This pull request sets up the initial backend project structure. The setup includes installing dependencies, setting up configurations, defining environment variables, and other initial configurations needed to start the development of the backend API with Auth and User Services.

### Changes:

### Initialize Project:
- Initialized a new Node.js project using npm init.
- Installed necessary dependencies such as KoaJS, Knex, SQLite3, JWT, dotenv, and other related packages.

### Project Structure:
- Created the basic folder structure including src, config, controllers, models, routes, and migrations.
- Added initial files for Koa server setup.

### Environment Configuration:
- Set up dotenv to manage environment variables.
- Created a .env file with required environment variables.

### Database Setup:
- Installed and configured Knex for database migrations and interactions.
- Created initial migration files for the users and migrations tables.

### Basic Server Setup:
- Set up the Koa server with basic middleware (logger, body parser, etc.).
- Added a health check endpoint to ensure the server is running.

### Documentation:
- Added a README.md file with installation instructions, environment setup, and project details.

### Steps to Test:

1. Clone the repository and checkout to the branch associated with this pull request.
2. Install the dependencies using pnpm install.
3. Set up the environment variables as described in the README.md.
4. Run the initial migrations using pnpx knex migrate:latest.
5. Start the server using pnpm start.
6. Verify that the server is running by accessing the health check endpoint at http://localhost:3000/health.
